### PR TITLE
Added support for boolean exclusive disjunction in standard library

### LINKED ISSUE #1096: Missing XOR support for booleans
**Is your feature request related to a problem? Please describe.**
The old QDK had a handy [Xor](https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.logical.xor) function which could be used to perform XOR between two booleans. 

**Describe the solution you'd like**
It was a super simple function but very useful, so it could just be added back

```qsharp
    function Xor(a : Bool, b : Bool) : Bool {
        (a or b) and ((not a) or (not b))
    }
```

Alternatively the `^^^` (or `^`) could be extended to work with booleans - at the moment it works with ints only. For example in C# I could write `true ^ true` or `false ^ true`

**Describe alternatives you've considered**
1. do nothing and force everyone to write this by hand
2. add the function
3. extend the operator

**Additional context**
None

### PR DESCRIPTION
**Pull request by filipw**
Fixes #1096 

The motivation is discussed in that issue as well. I put in the `Microsoft.Quantum.Logical` namespace, but we can move it into a different one if needed.

### CODE REVIEW

File: library/std/logical.qs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    /// A `Bool` which is `true` if and only if exactly one of `first` and `second` is `true`.
+    ///
+    /// # Example
+    /// ```qsharp
+    ///
+    /// let result = Xor(true, false);
+    /// // result is true
+    /// ```
+    function Xor(first : Bool, second : Bool) : Bool {
+        (first or second) and ((not first) or (not second))
```
Comment (msoeken) [inline library/std/logical.qs:26]:
```suggestion
        first != second
```

**Review by msoeken — CHANGES_REQUESTED** [review]
I suggest to simplify the expression.

File: library/std/logical.qs (hunk 1 above)
Comment (filipw) [reply library/std/logical.qs]: yes makes sense, thanks!

**Review by msoeken — APPROVED** [review]
Looks good to me. Thanks @filipw

**Review by DmitryVasilevsky — APPROVED** [review]
Thank you! We decided to go with the compatibility argument.

**Pull request merged**